test/utilities/notifications.ts

```diff
@@ -5,137 +5,70 @@
  */
 
-import { Workbench } from 'wdio-vscode-service';
-import { Duration, log, pause } from './miscellaneous.ts';
+import { Notification } from 'wdio-vscode-service';
+import { Duration, log } from './miscellaneous.ts';
 import { getWorkbench } from './workbench.ts';
 
 export async function waitForNotificationToGoAway(
-  workbench: Workbench,
   notificationMessage: string,
-  durationInSeconds: Duration,
-  matchExactString: boolean = true
+  durationInSeconds: Duration
 ): Promise<void> {
-  await pause(Duration.seconds(5));
-  const startDate = new Date();
-  while (true) {
-    const notificationWasFound = await notificationIsPresent(
-      workbench,
-      notificationMessage,
-      matchExactString
-    );
-    if (!notificationWasFound) {
-      return;
-    }
-
-    const currentDate = new Date();
-    const secondsPassed = Math.abs(currentDate.getTime() - startDate.getTime());
-    if (secondsPassed >= durationInSeconds.milliseconds) {
-      throw new Error(
-        `Exceeded time limit - notification "${notificationMessage}" is still present`
-      );
-    }
-  }
+  await findNotification(notificationMessage, false, durationInSeconds);
 }
 
-export async function notificationIsPresent(
-  workbench: Workbench,
-  notificationMessage: string,
-  matchExactString: boolean = true
-): Promise<boolean> {
-  const notifications = await workbench.getNotifications();
-  for (const notification of notifications) {
-    const message = await notification.getMessage();
-    if (matchExactString) {
-      if (message === notificationMessage) {
-        return true;
-      }
-    } else {
-      if (message.startsWith(notificationMessage)) {
-        return true;
-      }
-    }
-  }
+export async function notificationIsPresent(notificationMessage: string): Promise<boolean> {
+  const notification = await findNotification(notificationMessage, true, Duration.seconds(1));
 
-  return false;
+  return notification ? true : false;
 }
 
 export async function notificationIsPresentWithTimeout(
-  workbench: Workbench,
   notificationMessage: string,
-  durationInSeconds: Duration,
-  matchExactString: boolean = true
+  durationInSeconds: Duration
 ): Promise<boolean> {
-  const startDate = new Date();
-  let currentDate: Date;
-  let secondsPassed: number = 0;
+  const notification = await findNotification(notificationMessage, true, durationInSeconds);
 
-  // Keep on searching for the notification until it is found or the timeout is reached
-  while (secondsPassed < durationInSeconds.seconds) {
-    // Get a list of all the notifications that are currently present
-    const notifications = await workbench.getNotifications();
+  return notification ? true : false;
+}
 
-    // If there are no notifications present, wait 3 seconds before trying again
-    if (notifications.length === 0) {
-      await pause(Duration.seconds(3));
-    }
+export async function notificationIsAbsent(notificationMessage: string): Promise<boolean> {
```

Comment:
this logic+naming is killing me 💀 do we really need this in addition to notificationIsPresent?